app/(dashboard)/enroll/[repositoryId]/page.tsx

```diff
@@ -1,90 +1,41 @@
+"use client";
+
 import { DashboardHeader } from "@/components/header";
 import { DashboardShell } from "@/components/shell";
 import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
-import Image from "next/image";
-import Link from "next/link";
+import { TRepository, ZRepository } from "@/types/repository";
+import { useEffect, useState } from "react";
 
+import { getRepositoryDataAction } from "./actions";
 import EnrollmentStatusBar from "./enrollmentStatusBar";
-import FB1 from "./fb-view-1.webp";
-import FB2 from "./fb-view-2.webp";
 
-const frombricksFeatures = [
-  {
-    icon: "📲",
-    description: "Create conversion-optimized surveys with our no-code editor with several question types.",
-  },
-  {
-    icon: "📚",
-    description: "Choose from a variety of best-practice templates.",
-  },
-  {
-    icon: "👩🏻",
-    description:
-      "Launch and target your surveys to specific user groups without changing your application code.",
-  },
-  {
-    icon: "🔗",
-    description: "Create shareable link surveys.",
-  },
-  {
-    icon: "👨‍👩‍👦",
-    description: "Invite your team members to collaborate on your surveys.",
-  },
-  {
-    icon: "🔌",
-    description: "Integrate Formbricks with Slack, Notion, Zapier, n8n, and more.",
-  },
-  {
-    icon: "🔒",
-    description: "All open source, transparent, and self-hostable.",
-  },
-];
+export default function RepositoryDetailPage({ params }) {
+  const [repository, setRepository] = useState<TRepository | null>(null);
 
-const formbricksTechStack = [
-  {
-    icon: "💻",
-    name: "Typescript",
-    href: "https://www.typescriptlang.org/",
-  },
-  {
-    icon: "🚀",
-    name: "Next.js",
-    href: "https://nextjs.org/",
-  },
-  {
-    icon: "⚛️",
-    name: "React",
-    href: "https://reactjs.org/",
-  },
-  {
-    icon: "🎨",
-    name: "TailwindCSS",
-    href: "https://tailwindcss.com/",
-  },
-  {
-    icon: "📚",
-    name: "Prisma",
-    href: "https://www.prisma.io/",
-  },
-  {
-    icon: "🔒",
-    name: "Auth.js",
-    href: "https://auth0.com/",
-  },
-  {
-    icon: "🧘‍♂️",
-    name: "Zod",
-    href: "https://github.com/colinhacks/zod",
-  },
-];
+  useEffect(() => {
```

Comment:
@mattinannt has a better answer ;)

Reply:
For this page I don't see why the page needs to be client side. It would be much easier to make it serverside and fetch data directly from the service without action and useEffect

For the React Query/SWR question: they are not really built and suitable for server components and server actions. We would need to build our own API endpoints for that.